NEW: Add rotation parameter to Gaussian2D component

magnunor opened the pull request:
I added a rotation parameter to the Gaussian2D model based on the general equation here https://en.wikipedia.org/wiki/Gaussian_function#Two-dimensional_Gaussian_function

francisco-dlp committed:
Split Model in BaseModel and Model1D and add Model2D scratch

francisco-dlp committed:
Add Gaussian2D draft

dnjohnstone committed:
2D model first working implementation

dnjohnstone committed:
Merge branch 'master' into New_Model2D

Conflicts:
	hyperspy/component.py
	hyperspy/hspy.py
	hyperspy/model.py
	hyperspy/models/eelsmodel.py
	hyperspy/signal.py

dnjohnstone committed:
Adding 2D components (unfinished)Adding 2D components (unfinished)

dnjohnstone committed:
Fixing issues from merging master

dnjohnstone committed:
Updating to use ModelComponents

magnunor committed:
NEW: Add rotation parameter to Gaussian2D component

vidartf commented on hyperspy/_2d_components/gaussian2d.py:
```diff
@@ -73,9 +79,21 @@ def function(self, x, y):
+        temp_2sy2 = 2*sy**2
+        temp_cos2_theta = math.cos(theta)**2
+        temp_sin2_theta = math.sin(theta)**2
+        temp_sin_theta2 = math.sin(2*theta)
```
'temp' prefix feels redundant :)

magnunor committed:
Gaussian2D: rename internal variables, add properties

Added properties: ellipticity, fwhm_x, fwhm_y, rotation_degrees
Also setters for fwhm_x and fwhm_y. Removed temp_ prefixes to
internal variables in .function()

magnunor commented:
I removed the "temp_" prefixes and added properties fwhm, rotation_degrees and ellipticity.
